tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -98,6 +121,11 @@ public void CloseSpan(Span span)
                 Profiler.Instance.ContextTracker.SetEndpoint(span.RootSpanId, span.ResourceName);
             }
 
+            if (IastInitialized)
+            {
+                IastRequestContext?.AddIastTagsIfNeeded(span);
```

Comment:
This only adds tags is `span` is the root span and it's type is `web`, so we can probably move it into the `if (span.IsRootSpan && span.Type == SpanTypes.Web)` a few lines above and remove the second check inside `AddIastTagsIfNeeded()`.